lib/ui/mockData/MockRecordAmountViewModel.dart

```diff
@@ -0,0 +1,149 @@
+
+class MockRecordAmountViewModel {
+  static RecordAmountViewModel buildSale() {
+    List<RecordAmountListItemViewModel> list = [];
+
+    for (var i = 0; i < 3; i++) {
+      list.add(MockRecordAmountListItemViewModel.build(i));
+    }
+
+    return RecordAmountViewModel(
+        loadingStatus: LoadingStatus.SUCCESS,
+        actions: list,
+        buttonTitle: _mockButtonTitle[1],
+        onTap: null,
+        type: RecordType.sale);
+  }
+
+  static RecordAmountViewModel buildCost() {
+    List<RecordAmountListItemViewModel> list = [];
+
+    for (var i = 0; i < 3; i++) {
+      list.add(MockRecordAmountListItemViewModel.build(i));
+    }
+
+    return RecordAmountViewModel(
+        loadingStatus: LoadingStatus.SUCCESS,
+        actions: list,
+        buttonTitle: _mockButtonTitle[0],
+        onTap: null,
+        type: RecordType.cost);
+  }
+
+  static RecordAmountViewModel buildViewSale() {
+    List<RecordAmountListItemViewModel> list = [];
+
+    for (var i = 0; i < 3; i++) {
+      list.add(MockRecordAmountListItemViewModel.buildNotEditable(i));
+    }
+
+    return RecordAmountViewModel(
+      loadingStatus: LoadingStatus.SUCCESS,
+      actions: list,
+      amount: "12.48",
+      buttonTitle: _mockButtonTitle[1],
+      onTap: null,
+      type: RecordType.sale,
+      isEditable: false,
+    );
+  }
+
+  static RecordAmountViewModel buildViewCost() {
+    List<RecordAmountListItemViewModel> list = [];
+
+    for (var i = 0; i < 3; i++) {
+      list.add(MockRecordAmountListItemViewModel.buildNotEditable(i));
+    }
+
+    return RecordAmountViewModel(
+        loadingStatus: LoadingStatus.SUCCESS,
+        actions: list,
+        buttonTitle: _mockButtonTitle[1],
+        onTap: null,
+        type: RecordType.cost,
+        isEditable: false,
+        amount: "233.3");
+  }
+}
+
+class MockRecordAmountListItemViewModel {
+  static RecordAmountListItemViewModel build(int index) {
+    return RecordAmountListItemViewModel(
+      icon: _mockItemIcon[index],
+      title: _mockItemTitle[index],
+      hint: _mockItemHint[index],
+      arrow: _mockArrowIcon[index],
+      selectedDate: _mockSelectedDate[index],
+      listOfCrops: _mockListOfCrops[index],
+      listener: null,
+    );
+  }
+
+  static RecordAmountListItemViewModel buildNotEditable(int index) {
+    return RecordAmountListItemViewModel(
+      icon: _mockItemIcon[index],
+      title: _mockItemTitle[index],
+      hint: _mockItemHint[index],
+      arrow: _mockArrowIcon[index],
+      description: _mockDescription.random(),
+      selectedDate: _mockSelectedDate[index],
+      listOfCrops: _mockListOfCrops[index],
+      selectedItem: _mockCrop[index],
+      listener: null,
+      isEditable: false,
+    );
+  }
+}
+
+List _mockCrop = [
+  "Okra",
+  "Tomatoes",
+  "Potatoes",
+  "Cowpeas",
+  "Sweetcorn",
+  "Cucumber",
+  "Beetroot"
+];
+
+List _mockSelectedDate = [DateTime.now(), null, null];
```

Comment:
I have A new MockDateClass that can help here.

Reply:
Perfect, I'll implement it.